ceph_deploy/hosts/remotes.py

```diff
@@ -153,6 +153,10 @@ def remove_whitespace_from_assignments():
 def write_conf(cluster, conf, overwrite):
     """ write cluster configuration to /etc/ceph/{cluster}.conf """
     path = '/etc/ceph/{cluster}.conf'.format(cluster=cluster)
+    # Stateless OS like ClearLinux does not have anything populated in /etc.
+    # Create the folder before we write the config file.
+    if not os.path.isdir('/etc/ceph'):
+        os.makedirs('/etc/ceph', mode=0o700)
```

Comment:
I think we should rewrite this function, because at the end of the function there's a check and an exception if /etc/ceph doesn't exists, which is nonsense now.
I'm not sure I can commit to this branch so here goes mi proposal:

```python
def write_conf(cluster, conf, overwrite):
    """ write cluster configuration to /etc/ceph/{cluster}.conf """
    path = '/etc/ceph/{cluster}.conf'.format(cluster=cluster)

    # Stateless OS like ClearLinux does not have anything populated in /etc.
    # Create the folder before we write the config file.
    if not os.path.isdir('/etc/ceph'):
        os.makedirs('/etc/ceph', mode=0o700)

    if os.path.exists(path):
        tmp_file = tempfile.NamedTemporaryFile('w', dir='/etc/ceph', delete=False)
        err_msg = 'config file %s exists with different content; use ' \
                  '--overwrite-conf to overwrite' % path
        with open(path, 'r') as f:
            old = f.read()
            if old != conf and not overwrite:
                raise RuntimeError(err_msg)
        tmp_file.write(conf)
        tmp_file.close()
        shutil.move(tmp_file.name, path)
    else:
        with open(path, 'w') as f:
            f.write(conf)
    os.chmod(path, 0o644)
```

Reply:
@HacKanCuBa  thnks for the tip. Change integrated.

Reply:
@HacKanCuBa  thanks. done.